internal/subscriber/model.go

```diff
@@ -81,16 +84,32 @@ type AckMessage struct {
 const ackIDSeparator = "_"
 
 // NewAckMessage ...
-func NewAckMessage(subscriberID, topic string, partition, offset, deadline int32, messageID string) IAckMessage {
-	// TODO: add needed validations on all fields
+func NewAckMessage(subscriberID, topic string, partition, offset, deadline int32, messageID string) (IAckMessage, error) {
+
+	//Validating parameters for illegal values
+	if partition < 0 {
+		err := errors.New("partition value cannot be less than 0")
```

Comment:
move these error var's out of the function to the top of the file. You can then assert these in your UTs.

Reply:
I can assert the errors in the test this way also right? As we are returning err from the function, I can use the return values to assert the returned error. Is there any added benefits of moving the err variable to the top of the file?

Reply:
Two things:
1. If you declare all the errors in the top of the file, it lets everyone know all the kind of errors that your code throws.
2. It looks a little neater (personal opinion)
3. Asserting the same error var in test cases ensures to a good extent that, any modifications in future w.r.t the same error gets reflected in test cases as well.. no need to change the test cases.

Reply:
Okay Got it. Is there any file in which this is followed that I can use as a reference?

Reply:
https://github.com/razorpay/metro/blob/a494477e00cd8b976e3e2569a0df527fecdcf317/internal/subscription/validation.go#L62